lib/src/util/windows/wsa.cc

```diff
@@ -40,8 +40,12 @@ namespace facter { namespace util { namespace windows {
 
     string wsa::saddress_to_string(SOCKET_ADDRESS const& addr) const
     {
-        DWORD size = INET6_ADDRSTRLEN;
-        wchar_t buffer[INET6_ADDRSTRLEN];
+        if (!addr.lpSockaddr) {
+            return {};
+        }
+
+        DWORD size = INET6_ADDRSTRLEN+1;
+        wchar_t buffer[INET6_ADDRSTRLEN+1];
```

Comment:
FWIW, it looks like [GetNameInfoW](https://msdn.microsoft.com/en-us/library/windows/desktop/ms738531\(v=vs.85\).aspx) could provide you with more detail about potential errors if you want to do more validation of the address.